lsp-clients.el

```diff
@@ -828,5 +830,120 @@ responsiveness at the cost of possible stability issues."
+  (map 'list
+   (lambda (x)
+     (concat
+      (expand-file-name
+       lsp-rf-language-server-dir)
+      x
+      ".json"))
+   dirs))
+
+(defun parse-rf-language-server-globs-to-regex (vector)
+  "Translates vector of globs to regex (as a group separated by or statements)"
+  (let ((globs (if (eq vector [])
+                        ["*.robot" "*.resource"]
+                      vector)))
+    (concat "\\("(mapconcat 'eshell-glob-regexp globs "\\|") "\\)")))
+
+;; included files are extracted in lsp-register-client function
+;; using the glob-to-regex function above as a third argument in
+;; the directory-files-recursively function
+
+(defun rf-language-server-exclude-paths (seq)
+  "Uses lsp-rf-language-server-exclude-paths vector of globs to
+delete matched files from sequence"
+  (cl-delete-if (lambda (x) (string-match-p
+                             (parse-rf-language-server-globs-to-regex
+                              lsp-rf-language-server-exclude-paths)
+                             x))
+                seq))
+
+(lsp-register-custom-settings
+ '(
+   ("rfLanguageServer.trace.server" lsp-rf-language-server-trace-server)
+   ("rfLanguageServer.logLevel" lsp-rf-language-server-log-level)
+   ("rfLanguageServer.libraries" lsp-rf-language-server-libraries)
+   ("rfLanguageServer.pythonKeywords" lsp-rf-language-server-python-keywords t)
+   ("rfLanguageServer.excludePaths" lsp-rf-language-server-exclude-paths)
+   ("rfLanguageServer.includePaths" lsp-rf-language-server-include-paths)
+   ))
+
+(lsp-register-client
+ (make-lsp-client :new-connection (lsp-stdio-connection
+                                   `("node" ,(expand-file-name
+                                              lsp-rf-language-server-main-file)))
+                  :major-modes '(robot-mode)
+                  :server-id 'rf-intellisense
+                  ;; :library-folders-fn (lambda (_workspace)
+                  ;;                        lsp-rf-language-server-libraries)
+                  :library-folders-fn (lambda (_workspace)
+                                         (parse-rf-language-server-library-dirs
+                                         lsp-rf-language-server-libraries))
+                  :initialized-fn (lambda (workspace)
+                                    (with-lsp-workspace workspace
+                                      (lsp--set-configuration
+                                       (lsp-configuration-section "rfLanguageServer"))
+                                      (lsp-request "buildFromFiles"
+                                                   (list :files
+                                                         (vconcat
+                                                          (rf-language-server-exclude-paths
+                                                           (directory-files-recursively
+                                                            (projectile-project-root)
```

Comment:
You should not use projectile here but `lsp--workspace-root`

Reply:
I've added lsp-workspace-root. I looked at lsp--workspace-root but I am not familiar with cl structures yet. I don't know what to use as argument (CL-X). I tried lsp--cur-workspace but its value is nil